api/server/router/container/backend.go

```diff
@@ -43,7 +43,7 @@ type stateBackend interface {
 	ContainerStop(name string, seconds int) error
 	ContainerUnpause(name string) error
 	ContainerUpdate(name string, hostConfig *container.HostConfig) ([]string, error)
-	ContainerWait(name string, timeout time.Duration) (int, error)
+	ContainerWait(name string, timeout time.Duration, waitForRemoval bool) (int, error)
```

Comment:
@mlaventure I don't think it will reduce the complexity.  Different code path due to having TTY or not is still necessary, because if with TTY or restart policy, we can't wait container to stop or remove because it could be detached but not stopped, at least the timeout is necessary.

The only benefit I'm seeing is we can save the extra arguments of `Wait`, do not introduce extra unnecessary options was also what I was trying to do.

So if you feels this is worth a try, maybe I should provide another implementation based on this?